[BEAM-2937] Add Combine Grouped Values URN.

youngoli opened the pull request:
I added a new URN (Combine Grouped Values) to the following doc, adding it in code as well.
https://s.apache.org/beam-runner-api-combine-model

Keeping the old URNs to avoid breaking existing code.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

youngoli commented:
R: @lukecwik @robertwb 

Whoever gets to it first.

youngoli commented:
Whoops, looks like the URN I added already exists. Didn't actually check for that before, I just assumed it would work.

Deleting my comment adding reviewers until I figure out how to fix this, then I'll add reviewers again.

youngoli commented:
Since the URN already exists, and is being used for nearly the same thing I need it for, the only change I really need to make right now is the comment on the URN. Also updated the comments for the other Combine URNs, adding the expected Payload accompanying the URN, in order to make them more consistent with the existing comments.

lukecwik commented on model/pipeline/src/main/proto/beam_runner_api.proto:
```diff
@@ -228,9 +228,9 @@ message StandardPTransforms {
-    // each of CombineComponents.
+    // Represents the Combine Grouped Values transform, as described in the
+    // following document:
+    // https://s.apache.org/beam-runner-api-combine-model#heading=h.aj86ew4v1wk
```
Should this be moved from `Composites` to `CombineComponents`?

youngoli replied:
Yeah, that seems like a good idea. Is it as easy as just deleting it and changing the numbers on the existing enum values, or would that cause issues?

youngoli commented:
I moved the CombineValues URN from Composites to CombineComponents. I also changed the numbers on the existing values enum values. Normally that's really bad with protos, but in this situation the enum in question is never actually used in any proto messages. They're exclusively for use by devs for fetching the URN strings from the beam_urn extension, so changing the values shouldn't affect anything.

youngoli commented:
Run Java PreCommit

youngoli commented:
Run Java PreCommit
